src/sidebar/components/moderation-banner.js

```diff
@@ -1,67 +1,113 @@
-  };
+const annotationMetadata = require('../annotation-metadata');
+const useStore = require('../store/use-store');
+const { withServices } = require('../util/service-context');
 
-  this.isHidden = function() {
-    return self.annotation.hidden;
-  };
+function ModerationBanner({ annotation, api, flash }) {
+  // actions
+  const store = useStore(store => ({
+    hide: store.hideAnnotation,
+    unhide: store.unhideAnnotation,
+  }));
 
-  this.isHiddenOrFlagged = function() {
-    const flagCount = self.flagCount();
-    return flagCount !== null && (flagCount > 0 || self.isHidden());
-  };
+  const flagCount = annotationMetadata.flagCount(annotation);
 
-  this.isReply = function() {
-    return annotationMetadata.isReply(self.annotation);
+  const isHiddenOrFlagged = function() {
+    return flagCount !== null && (flagCount > 0 || annotation.hidden);
   };
 
   /**
    * Hide an annotation from non-moderator users.
    */
-  this.hideAnnotation = function() {
+  const hideAnnotation = () => {
     api.annotation
-      .hide({ id: self.annotation.id })
-      .then(function() {
-        store.hideAnnotation(self.annotation.id);
+      .hide({ id: annotation.id })
+      .then(() => {
+        store.hide(annotation.id);
       })
-      .catch(function() {
+      .catch(() => {
         flash.error('Failed to hide annotation');
       });
   };
 
   /**
    * Un-hide an annotation from non-moderator users.
    */
-  this.unhideAnnotation = function() {
+  const unhideAnnotation = () => {
     api.annotation
-      .unhide({ id: self.annotation.id })
-      .then(function() {
-        store.unhideAnnotation(self.annotation.id);
+      .unhide({ id: annotation.id })
+      .then(() => {
+        store.unhide(annotation.id);
       })
-      .catch(function() {
+      .catch(() => {
         flash.error('Failed to unhide annotation');
       });
   };
-}
 
-/**
- * Banner shown above flagged annotations to allow moderators to hide/unhide the
- * annotation from other users.
- */
+  /**
+   * Creates the hide / unhide toggle button
+   */
+  let toggleButton;
+  if (annotation.hidden) {
+    toggleButton = (
+      <button
+        onClick={unhideAnnotation}
+        title="Make this annotation visible to everyone"
+      >
+        Unhide
+      </button>
+    );
+  } else {
+    toggleButton = (
+      <button
+        onClick={hideAnnotation}
+        title="Hide this annotation from non-moderators"
+      >
+        Hide
+      </button>
+    );
+  }
+
+  const bannerClasses = classnames({
+    'moderation-banner': true,
+    'is-flagged': flagCount > 0,
+    'is-hidden': annotation.hidden,
+    'is-reply': annotationMetadata.isReply(annotation),
+  });
+
+  /**
+   * Banner allows moderators to hide/unhide the
+   * flagged annotation(s) from other users.
+   */
+  return (
+    <span>
+      {isHiddenOrFlagged() && (
+        <div className={bannerClasses}>
+          {flagCount && !annotation.hidden && (
```

Comment:
I caught this bug -- didnt quite convert this right, its not calling the function `flagCount` its just checking that var exists and it always would since its a method... fixed!